### ISSUE TITLE
Entities card row type 'button' is not accepted

### ISSUE THREAD

**Issue by danielbrunt57** [home-assistant/frontend#9939]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The [entities card documentation](https://www.home-assistant.io/lovelace/entities/) lists `button` as a row type but it is not accepted.  I've copied/pasted the last code sample and it does not work until I remove the two `button` rows.

### Describe the behavior you expected

`- type: button` should work as documented.

### Steps to reproduce the issue

Add a card:
```
type: entities
title: Entities card sample
entities:
  - type: button
    icon: mdi:power
    name: Bed light transition
    action_name: Toggle light
    tap_action:
      action: call-service
      service: light.toggle
      service_data:
        entity_id: light.bed_light
        transition: 10
```

### What version of Home Assistant Core has the issue?

core-2021.8.8, core-2021.9.1

### What was the last working version of Home Assistant Core?

<unknown>

### In which browser are you experiencing the issue with?

Latest Microsoft Edge browser

### Which operating system are you using to run this browser?

Windows 10

### Problem-relevant frontend configuration

```yaml
type: entities
title: Entities card sample
entities:
  - type: button
    icon: mdi:power
    name: Bed light transition
    action_name: Toggle light
    tap_action:
      action: call-service
      service: light.toggle
      service_data:
        entity_id: light.bed_light
        transition: 10
  - type: divider
  - type: attribute
    entity: sun.sun
    attribute: elevation
    name: Sun elevation
    prefix: "~"
    suffix: Units
  - type: conditional
    conditions:
      - entity: sun.sun
        state: above_horizon
    row:
      entity: sun.sun
      type: attribute
      attribute: azimuth
      icon: mdi:angle-acute
      name: Sun azimuth
  - type: section
    label: Section example
  - type: weblink
    name: Home Assistant
    url: https://www.home-assistant.io/
    icon: mdi:home-assistant
  - type: button
    name: Power cycle LibreELEC
    icon: mdi:power-cycle
    tap_action:
      action: call-service
      confirmation:
        text: Are you sure you want to restart?
      service: script.libreelec_power_cycle
```

**Comment by eth0up** [comment]
Hi @danielbrunt57,

Are you seeing behavior similar to [this forum post](https://community.home-assistant.io/t/buttons-in-entities-card-requires-useless-configuration-entries/319791/8)?

I am also having this issue with `- type: button`, as well as with `- type: call-service` when I attempt to add certain optional attributes. It seems as though attributes requirements aren't mapping as expected, and some extra requirements are thrown into the mix as a result - likely a bug. I experienced this issue when I edited existing working configs, it was only once I entered edit mode that the errors appeared and I could not save any changes. When I clicked Cancel in the editor dialog, the button rows that I previously configured worked just fine.

To get the button row to work as expected within the editor (I was using with scripts, like you), I had to add the following items to my config as null whenever I wasn't using them:
```
    entity: null
    view: null
    conditions: null
    url: null
    entities: null
    attribute: null
    service: null
    text: null
    name: null
```

If you remove the "icon" line such as in your _Power cycle LibreELEC_ entry, does the editor stop displaying errors? That's what I was seeing.. 

My config, with null values set to work around this error (I allow icon to be pulled directly from the script in this case):
```
type: entities
entities:
  - type: button
    entity: script.test_alert
    name: Alert Testing Script
    action_name: test it
    view: null
    conditions: null
    url: null
    entities: null
    attribute: null
    service: null
    text: null
    tap_action:
      action: toggle
      confirmation:
        text: Are you sure you want to test alert?
```


Your _Power cycle LibreELEC_ entry as-is also displays errors on my end. Here's a modified version of it that doesn't display errors for me:
```
  - type: button
    name: Power cycle LibreELEC
    icon: mdi:power-cycle
    action_name: test it
    view: null
    conditions: null
    url: null
    entities: null
    attribute: null
    service: null
    text: null
    entity: null
    tap_action:
      action: call-service
      confirmation:
        text: Are you sure you want to restart?
      service: script.libreelec_power_cycle
```

**Comment by danielbrunt57** [comment]
Yes, exactly.  I just now went to raw configuration editor and added the sample code snippet to a new view:
```
  - title: Test
    path: test
    badges: []
    cards:
      - type: entities
        title: Entities card sample
        entities:
          - type: button
            icon: mdi:power
            name: Bed light transition
            action_name: Toggle light
            tap_action:
              action: call-service
              service: light.toggle
              service_data:
                entity_id: light.bed_light
                transition: 10
          - type: divider
          - type: attribute
            entity: sun.sun
            attribute: elevation
            name: Sun elevation
            prefix: '~'
            suffix: Units
          - type: conditional
            conditions:
              - entity: sun.sun
                state: above_horizon
            row:
              entity: sun.sun
              type: attribute
              attribute: azimuth
              icon: mdi:angle-acute
              name: Sun azimuth
          - type: section
            label: Section example
          - type: weblink
            name: Home Assistant
            url: https://www.home-assistant.io/
            icon: mdi:home-assistant
          - type: button
            name: Power cycle LibreELEC
            icon: mdi:power-cycle
            tap_action:
              action: call-service
              confirmation:
                text: Are you sure you want to restart?
              service: script.libreelec_power_cycle
```
then saved it and the card displays correctly so It's the UI editor that is fubar.  
If I try and edit it using the UI I get:
```
Configuration errors detected:
Required key "entities.0.entity" is missing.
Required key "entities.0.view" is missing.
Required key "entities.0.view" is missing.
Required key "entities.0.view" is missing.
Required key "entities.0.conditions" is missing.
Required key "entities.0.url" is missing.
Required key "entities.0.entities" is missing.
Required key "entities.0.entity" is missing.
Required key "entities.0.attribute" is missing.
Required key "entities.0.service" is missing.
Required key "entities.0.text" is missing.
Required key "entities.6.entity" is missing.
Required key "entities.6.view" is missing.
Required key "entities.6.view" is missing.
Required key "entities.6.view" is missing.
Required key "entities.6.conditions" is missing.
Required key "entities.6.url" is missing.
Required key "entities.6.entities" is missing.
Required key "entities.6.entity" is missing.
Required key "entities.6.attribute" is missing.
Required key "entities.6.service" is missing.
Required key "entities.6.text" is missing.
Visual editor is not supported for this configuration:
Key "entities.0.type" is not expected or not supported by the visual editor.
Key "entities.0.action_name" is not expected or not supported by the visual editor.
The provided value for "entities.0" is not supported by the visual editor. We support (string) but received ({"type":"button","icon":"mdi:power","name":"Bed light transition","action_name":"Toggle light","tap_action":{"action":"call-service","service":"light.toggle","service_data":{"entity_id":"light.bed_light","transition":10}}}).
Key "entities.0.icon" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.action_name" is not expected or not supported by the visual editor.
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.icon" is not expected or not supported by the visual editor.
Key "entities.0.name" is not expected or not supported by the visual editor.
Key "entities.0.action_name" is not expected or not supported by the visual editor.
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.icon" is not expected or not supported by the visual editor.
Key "entities.0.name" is not expected or not supported by the visual editor.
Key "entities.0.action_name" is not expected or not supported by the visual editor.
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.icon" is not expected or not supported by the visual editor.
Key "entities.0.name" is not expected or not supported by the visual editor.
Key "entities.0.action_name" is not expected or not supported by the visual editor.
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.action_name" is not expected or not supported by the visual editor.
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.icon" is not expected or not supported by the visual editor.
Key "entities.0.name" is not expected or not supported by the visual editor.
Key "entities.0.action_name" is not expected or not supported by the visual editor.
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.icon" is not expected or not supported by the visual editor.
Key "entities.0.action_name" is not expected or not supported by the visual editor.
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.0.action_name" is not expected or not supported by the visual editor.
Key "entities.0.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.0.type" is not supported by the visual editor. We support (custom element type) but received ("button").
Key "entities.6.type" is not expected or not supported by the visual editor.
The provided value for "entities.6" is not supported by the visual editor. We support (string) but received ({"type":"button","name":"Power cycle LibreELEC","icon":"mdi:power-cycle","tap_action":{"action":"call-service","confirmation":{"text":"Are you sure you want to restart?"},"service":"script.libreelec_power_cycle"}}).
Key "entities.6.icon" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.name" is not expected or not supported by the visual editor.
Key "entities.6.icon" is not expected or not supported by the visual editor.
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.name" is not expected or not supported by the visual editor.
Key "entities.6.icon" is not expected or not supported by the visual editor.
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.name" is not expected or not supported by the visual editor.
Key "entities.6.icon" is not expected or not supported by the visual editor.
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.name" is not expected or not supported by the visual editor.
Key "entities.6.icon" is not expected or not supported by the visual editor.
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.icon" is not expected or not supported by the visual editor.
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (literal) but received ("button").
Key "entities.6.tap_action" is not expected or not supported by the visual editor.
The provided value for "entities.6.type" is not supported by the visual editor. We support (custom element type) but received ("button").
You can still edit your config in YAML.
```
I can edit it in YAML mode but cannot save it due to the erroneous errors re: - `type: button`

**Comment by bramkragten** [comment]
Try 2021.9 first before opening an issue

**Comment by danielbrunt57** [comment]
I updated to 2021.9 this morning which was after posting...

**Comment by bramkragten** [comment]
Works here

![image](https://user-images.githubusercontent.com/5662298/131975005-86e5d086-ebf7-4240-92e3-86d86441c441.png)

**Comment by danielbrunt57** [comment]
What code do you have?

**Comment by danielbrunt57** [comment]
![image](https://user-images.githubusercontent.com/49846893/131976224-8b8ec1c2-e9b2-483d-b33c-5d5e69333a99.png)

**Comment by danielbrunt57** [comment]
Never mind, I pasted the code sample again from [](https://www.home-assistant.io/lovelace/entities/) into a new card it's working now.
![image](https://user-images.githubusercontent.com/49846893/131976676-36aac7ee-1117-409a-9f18-217b58353d24.png)
All I've done is some template editing and restarted HA a couple of times...
